categorization/bin/llm_categorization.py

```diff
@@ -1,22 +1,121 @@
+from datetime import datetime
 import json
-from typing import Dict, Any
+from typing import Dict, Any, Union
 from langchain_community.chat_models import ChatOllama
 from langchain_core.prompts import PromptTemplate
 
 
+def SexLevel(result_dict: Dict[str, str], r: str, key: str) -> None:
+    value = result_dict[key]
+    var1: str = ""
+    var2: str = ""
+
+    if "1" in value or "2" in value:
+        var1 = "1"
+        var2 = "2"
+    elif "0" in value or "1" in value:
+        var1 = "0"
+        var2 = "1"
+    elif "M" in value or "F" in value:
+        var1 = "M"
+        var2 = "F"
+    elif "m" in value or "f" in value:
+        var1 = "m"
+        var2 = "f"
+    elif "male" in value or "female" in value:
+        var1 = "male"
+        var2 = "female"
```

Comment:
Did you include something like `.lower()` in the code for the sex levels? If so, nevermind - if not - maybe we should to capture all the valid formats of the [BIDS sex column specification](https://bids-specification.readthedocs.io/en/stable/glossary.html#objects.columns.sex). And maybe include the "Other" parts also already? - I should include this in the parsing also ;)

Reply:
Yes .lower() will be better!! :smile: 
I will add it and the others section and ping you!